versioned/transfer/src/main/java/org/projectnessie/versioned/transfer/ImportPersistV23.java

```diff
@@ -129,4 +151,26 @@ void processCommit(Commit commit) throws ObjTooLargeException {
 
     importer.progressListener().progress(ProgressEvent.COMMIT_WRITTEN);
   }
+
+  @Override
+  void processGeneric(TransferTypes.GenericObj genericObj) throws ObjTooLargeException {
+    ObjType type = objTypeByName(genericObj.getTypeName());
+    ObjId id = objIdFromBytes(genericObj.getId());
+
+    Obj obj;
+    if (type.equals(StandardObjType.UNIQUE)) {
+      obj = UniqueIdObj.uniqueId(id, genericObj.getSpace(), genericObj.getData());
+    } else {
+      ByteBuffer data = genericObj.getData().asReadOnlyByteBuffer();
+      String versionToken = genericObj.hasVersionToken() ? genericObj.getVersionToken() : null;
+
+      obj =
+          SmileSerialization.deserializeObj(
```

Comment:
Do we want to check the `GenericObj.encoding` value?

Reply:
added